src/Workspaces/Core/Portable/FindSymbols/SearchQuery.cs

```diff
@@ -36,7 +36,7 @@ private SearchQuery(string name, SearchKind kind)
                     // its 'AreSimilar' method. That way we only create the WordSimilarityChecker
                     // once and it can cache all the information it needs while it does the AreSimilar
                     // check against all the possible candidates.
-                    var editDistance = new WordSimilarityChecker(name, substringsAreSimilar: false);
+                    var editDistance = WordSimilarityChecker.Allocate(name, substringsAreSimilar: false);
```

Comment:
I'll comment.  It's not great, but it's non trivial to expand out disposal to the SearchQuery type as well.